src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java

```diff
@@ -442,7 +474,17 @@ public boolean useVariantAnnotations() {
      * Returns the name of the source of reads data. It can be a file name or URL.
      */
     protected String getReadSourceName(){
-        return readInput;
+        if (readInputs.size() > 1) {
+            throw new GATKException("Multiple ReadsDataSources specificed but a single source requested by the tool");
```

Comment:
not ideal maybe but I don't know what else to do about this method...  maybe change it to getReadSourceNames and return a list?

Reply:
Right, I wanted to avoid returning a list because there are a bunch of places in the code where tools expect there to be only one read source kicking around and I didn't want to risk breaking something or having to uproot everything... I agree its pretty gross... Theoretically it shouldn't be a problem for most tools which don't accept multiple inputs anyway

Reply:
I think we should change it to a list, or even get rid of it. All of the existing consumers except one use it to generate an output name for saving metrics (relatively easy to change); the other one (`getRecommendedNumReducers`) uses it to determine the number of reducers, which won't work correctly on multiple inputs. I think `getRecommendedNumReducers` should be updated either way.

Reply:
@cmnbroad Alright, I'll return a list. `getRecommendedNumReducers` has already been updated in this branch to sum over the all the read input files.